recipes/openimageio/all/conanfile.py

```diff
@@ -63,260 +64,250 @@ class OpenImageIOConan(ConanFile):
+        tc.variables["USE_FREETYPE"] = self.options.with_freetype
+        tc.variables["USE_LIBWEBP"] = self.options.with_libwebp
+        tc.variables["USE_OPENJPEG"] = self.options.with_openjpeg
 
-        if self.options.with_openvdb:
-            cmake.definitions["CMAKE_CXX_STANDARD"] = 14
-
-        cmake.configure(build_folder=self._build_subfolder)
-        return cmake
+        tc.generate()
+        cd = CMakeDeps(self)
+        cd.generate()
 
     def build(self):
-        self._patch_sources()
-
-        cmake = self._configure_cmake()
+        apply_conandata_patches(self)
+        cmake = CMake(self)
+        cmake.configure()
         cmake.build()
 
     def package(self):
-        cmake = self._configure_cmake()
+        copy(self, "LICENSE.md", src=self.source_folder, dst=os.path.join(self.package_folder, "licenses"))
+        cmake = CMake(self)
         cmake.install()
+        rmdir(self, os.path.join(self.package_folder, "share"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
+        rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
 
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "cmake"))
-        files.rmdir(self, os.path.join(self.package_folder, "lib", "pkgconfig"))
-        files.rmdir(self, os.path.join(self.package_folder, "share"))
+    @staticmethod
+    def _conan_comp(name):
+        return f"openimageio_{name.lower()}"
 
-        self.copy("LICENSE.md", src=self._source_subfolder, dst="licenses")
+    def _add_component(self, name):
+        component = self.cpp_info.components[self._conan_comp(name)]
+        component.set_property("cmake_target_name", f"OpenImageIO::{name}")
+        component.names["cmake_find_package"] = name
+        component.names["cmake_find_package_multi"] = name
+        return component
 
     def package_info(self):
         self.cpp_info.set_property("cmake_file_name", "OpenImageIO")
-        self.cpp_info.set_property("cmake_target_name", "OpenImageIO::OpenImageIO")
         self.cpp_info.set_property("pkg_config_name", "OpenImageIO")
 
-        self.cpp_info.components["openimageio_util"].set_property(
-            "cmake_target_name", "OpenImageIO::OpenImageIO_Util"
-        )
-        self.cpp_info.components["openimageio_util"].libs = ["OpenImageIO_Util"]
-        self.cpp_info.components["openimageio_util"].requires = [
+        self.cpp_info.names["cmake_find_package"] = "OpenImageIO"
+        self.cpp_info.names["cmake_find_package_multi"] = "OpenImageIO"
+        self.cpp_info.names["pkg_config"] = "OpenImageIO"
+
+        # OpenImageIO::OpenImageIO_Util
+        OpenImageIOUtil = self._add_component("OpenImageIO_Util")
+        OpenImageIOUtil.libs = ["OpenImageIO_Util"]
+        OpenImageIOUtil.requires = [
             "boost::filesystem",
             "boost::thread",
             "boost::system",
             "boost::regex",
+            "Imath::Imath",
```

Comment:
```suggestion
            "imath::imath",
```
To avoid the "ConanException: error generating context for 'openimageio/2.3.7.2': {'ref': Imath/unknown@unknown/unknown, 'build': False, 'direct': True, 'test': False, 'visible': True}" error.